core/lz4/src/ZipLZ4.cxx

```diff
@@ -24,9 +24,16 @@ typedef unsigned long long XXH64_hash_t;
 typedef struct {
    unsigned char digest[8];
 } XXH64_canonical_t;
+
+#ifdef BUILTIN_LZ4
 extern "C" XXH64_hash_t LZ4_XXH64(const void *input, size_t length, unsigned long long seed);
 extern "C" void LZ4_XXH64_canonicalFromHash(XXH64_canonical_t *dst, XXH64_hash_t hash);
 extern "C" XXH64_hash_t LZ4_XXH64_hashFromCanonical(const XXH64_canonical_t *src);
+#else
+extern "C" XXH64_hash_t XXH64(const void *input, size_t length, unsigned long long seed);
+extern "C" void XXH64_canonicalFromHash(XXH64_canonical_t *dst, XXH64_hash_t hash);
+extern "C" XXH64_hash_t XXH64_hashFromCanonical(const XXH64_canonical_t *src);
+#endif
```

Comment:
Hi Oksana, I think that here it would be better to have something like this:
```cpp
#ifdef BUILTIN_LZ4
#define XXH64 LZ4_XXH64
#define XXH64_canonicalFromHash LZ4_XXH64_canonicalFromHash
#define XXH64_hashFromCanonical LZ4_XXH64_hashFromCanonical
#endif
```
Then the code itself can use, e.g., `XXH64(input, length, seed);` without needing `#ifdefs`.

Reply:
@amadio, you are right, much more better!